stack.py

```diff
@@ -16,4 +16,21 @@ def pop(self):
     def top(self):
         return self.items[len(self.items) - 1]
 
+    def reverse1(self):
+        result = self.items.copy()
+        lo = 0
+        hi = len(self.items) - 1
+        while lo < hi:
+            result[lo], result[hi] = result[hi], result[lo]
+            lo += 1
+            hi -= 1
+        return result
```

Comment:
great! love the swapping method. I see you've decided to implement a non-mutating reverse :) 

I do have a question for you, however: if you were to create a function that _**does**_ mutate the` items` member variable, how would you go about doing so? :)